nest-tax-backend/src/card-payment-reporting/card-payment-reporting.service.ts

```diff
@@ -327,10 +328,17 @@ export class CardPaymentReportingService {
         if (!fileDate) return null
 
         const dateInfo = this.getDateInfo(fileDate)
+        if (dateInfo.today.isBefore(dayjs().subtract(180, 'day'))) {
```

Comment:
Make a variable out of `dayjs().subtract(180, 'day')` before the `Promise.all`, as it only needs to be computed once and then can be reused for all files in the loop. Also, with appropriate variable name it will increase readability.